src/classes/Addresses.cls

```diff
@@ -0,0 +1,1133 @@
+            Account acc = new Account(Id = accId);
+            if (Addresses.shouldUpdateAddress(accountsById.get(accId), addr)) {
+                NPSP_Address address = new NPSP_Address(addr);
+                NPSP_HouseholdAccount household = new NPSP_HouseholdAccount(acc);
+                household.billingAddress(address);
+                listAccUpdate.add(household.getRecord());
+            }
+
+            // check other addresses on this household that are no longer the default
+            if (mapAccIdListAddrHH != null && mapAccIdListAddrHH.size() > 0 && mapAccIdListAddrHH.get(accId) != null) {
+                for (Address__c addrHH : mapAccIdListAddrHH.get(accId)) {
+                    if (addrHH.Id == addrDefault.Id) {
+                        continue;
+                    }
+                    if (addrHH.Default_Address__c) {
+                        addrHH.Default_Address__c = false;
+                        addrHH.Latest_End_Date__c = System.today();
+                        listAddrUpdate.add(addrHH);
+                    }
+                }
+            }
+
+            // check the Contacts on this household
+            if (mapAccIdListCon != null && mapAccIdListCon.containsKey(accId)) {
+                for (Contact con : mapAccIdListCon.get(accId)) {
+                    NPSP_Contact npspContact = new NPSP_Contact(con);
+                    // detect that the contact's current address was deleted.
+                    // and if so clear any override flag so it will get the default address.
+                    if (npspContact.currentAddress() == null) {
+                        npspContact.getRecord().is_Address_Override__c = false;
+                    }
+                    NPSP_Address npspAddress = new NPSP_Address(addr);
+                    if (npspContact.hasAddressOverride() == false ||
+                            npspContact.currentAddress() == addr.Id) {
+                        if (Addresses.shouldUpdateAddress(con, addr)) {
+                            npspContact.currentAddress(npspAddress);
+                            listConUpdate.add(con);
+                        }
+                    }
+                }
+            }
+        }
+
+        if (listAccUpdate.size() > 0) {
+            dmlWrapper.objectsToUpdate.addAll((List<SObject>)listAccUpdate);
+        }
+        if (listAddrUpdate.size() > 0) {
+            dmlWrapper.objectsToUpdate.addAll((List<SObject>)listAddrUpdate);
+        }
+        if (listConUpdate.size() > 0) {
+            dmlWrapper.objectsToUpdate.addAll((List<SObject>)listConUpdate);
+        }
+
+        return mapAccIdHHInfo;
+    }
+
+    /*******************************************************************************************************
+    * @description helper class to keep track of the default and seasonal address for a household
+    */
+    public class HHInfo {
+        public Id accId;
+        public Address__c addrDefault;
```

Comment:
Would this eventually use NPSP_Address as a member instead of the SObject?

Reply:
Great question. This class actually looks like a data structure intended to hold a subset of the information that w should be in an NPSP_Account.  I would think you would want to access the default addresfor an account through a method like npspAccount.defaultAddress().  And I'm not exactly surhow this object is used with a single seasonal address since an account could have multiple seasonal addresses.  I don't know that I'll get into replacing this data structure in this work stream but it is a great callout and something to look at for potential future refactoring.